src/octopeer-github/main/Database/EventObject/EventObject.ts

```diff
@@ -0,0 +1,16 @@
+/// <reference path="KeystrokeEvent.ts"/>
+/// <reference path="MouseClickEvent.ts"/>
+/// <reference path="MousePositionEvent.ts"/>
+/// <reference path="MouseScrollEvent.ts"/>
+/// <reference path="SemanticEvent.ts"/>
+/// <reference path="WindowResolutionEvent.ts"/>
+/**
+ * Created by Maarten on 09-06-2016.
+ *
+ * One universal interface which is a mix of all EventObjects.
+ * The type should match the interface name of the data field.
+ */
+interface EventObject {
+    type: string;
+    data: KeystrokeEvent | MouseClickEvent | MousePositionEvent | MouseScrollEvent | SemanticEvent | WindowResolutionEvent;
```

Comment:
I didn't do exactly that, but something else that is close enough. When you have one interface that covers all `EventObjectData` interfaces, you cannot properly query the data anymore because not all fields are defined in all interfaces.